### PR TITLE
Feature/scrypto merge 2023 05 04

### PR DESCRIPTION
**Pull request by LukasGasior1**
Still a few TODOs left

### CODE REVIEW

**Review by jakrawcz-rdx — COMMENTED** [review]
Wow, nice simplifications happened in a drive-by, cool!

Some TODOs look severe... Can you somehow indicate which of them are planned to be addressed in this very PR (e.g. `TODO(wip)` or `TODO(this pr)`, and which ones are supposed to wait?

File: core-rust-bridge/src/main/java/com/radixdlt/genesis/GenesisDataChunk.java [hunk 1]
```diff
@@ -62,62 +62,66 @@
...
-public abstract class KnownGenesis {
-
-  private KnownGenesis() {
-    // only prevent instantiation
+public sealed interface GenesisDataChunk {
+  static void registerCodec(CodecMap codecMap) {
+    codecMap.registerForSealedClassAndSubclasses(
+        GenesisDataChunk.class,
+        (codecs) ->
+            EnumCodec.fromEntries(
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core-rust-bridge/src/main/java/com/radixdlt/genesis/GenesisDataChunk.java:84]: it should work 👍

File: core-rust/state-manager/src/mempool/mempool_manager.rs [hunk 2]
```diff
@@ -92,7 +92,7 @@ impl MempoolManager {
     /// Creates a manager and registers its metrics.
     pub fn new(
         mempool: Arc<RwLock<SimpleMempool>>,
-        relay_dispatcher: MempoolRelayDispatcher,
+        relay_dispatcher: Option<MempoolRelayDispatcher>,
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core-rust/state-manager/src/mempool/mempool_manager.rs:95]: 👍 added a separate constructor

File: core-rust/state-manager/src/state_manager.rs [hunk 3]
```diff
@@ -169,38 +174,42 @@ where
...
+            state_tracker.latest_transaction_identifiers(),
+            &hash,
             &self
                 .execution_configurator
                 .wrap(executable, ConfigType::Genesis)
-                .warn_after(TRANSACTION_RUNTIME_WARN_THRESHOLD, logged_description),
+                .warn_after(TRANSACTION_RUNTIME_WARN_THRESHOLD, &logged_description),
         );
-        let commit = processed.expect_commit(logged_description);
-        commit.check_success(logged_description);
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core-rust/state-manager/src/state_manager.rs:195]: we still do, in `execute_genesis`

File: core-rust/state-manager/src/store/rocks_db.rs [hunk 4]
```diff
@@ -719,9 +719,39 @@ impl QueryableProofStore for RocksDBStore {
...
+            .get_cf(self.cf_handle(&Substates), encoded_key_bytes)
+            .unwrap()
+    }
+
+    fn list_entries(
+        &self,
+        partition_key: &DbPartitionKey,
+    ) -> Box<dyn Iterator<Item = PartitionEntry> + '_> {
+        let partition_key = partition_key.clone();
+        let start = encode_to_rocksdb_bytes(&partition_key, &DbSortKey(vec![0]));
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core-rust/state-manager/src/store/rocks_db.rs:869]: good point; I followed the scrypto impl here

[the remark this answers is not in the document]
Comment (LukasGasior1) [reply core-rust/state-manager/src/store/rocks_db.rs:869]: changed to `vec![]` both here and in overlay db

File: core-rust/core-api-server/src/core_api/generated/models/state_resource_manager.rs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * The version of the OpenAPI document: 0.4.0
+ * 
+ * Generated by: https://openapi-generator.tech
+ */
+
+
+
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(tag = "resource_type")]
+pub enum StateResourceManager {
```
Comment (malexandru-rdx) [inline core-rust/core-api-server/src/core_api/generated/models/state_resource_manager.rs:15]: This is not on par with Java models (missing `resourceType`). Which also makes clippy complain about unused `to_api_resource_type`.

If I regenerate, indeed the only file modified is this.

**Pull request merged**